test/unit/lib/plugins/aws/invoke.test.js

```diff
@@ -454,7 +456,143 @@ describe.skip('test/unit/lib/plugins/aws/invoke.test.js', () => {
     // https://github.com/serverless/serverless/blob/537fcac7597f0c6efbae7a5fc984270a78a2a53a/test/unit/lib/plugins/aws/invoke.test.js#L269-L287
   });
 
-  it('TODO: should fail the process for failed invocations', async () => {
+  it('should support `--context` param', async () => {
+    const lambdaInvokeStub = sinon.stub();
+
+    const result = await runServerless({
+      fixture: 'invocation',
+      cliArgs: ['invoke', '--function', 'callback', '--context', 'somecontext'],
+      awsRequestStubMap: {
+        Lambda: {
+          invoke: args => {
+            lambdaInvokeStub.returns('payload');
+            return lambdaInvokeStub(args);
+          },
+        },
+      },
+    });
+    expect(lambdaInvokeStub).to.have.been.calledOnce;
+    expect(lambdaInvokeStub.args[0][0]).to.deep.equal({
+      ClientContext: 'InNvbWVjb250ZXh0Ig==', // "somecontext"
+      FunctionName: result.serverless.service.getFunction('callback').name,
+      InvocationType: 'RequestResponse',
+      LogType: 'None',
+      Payload: Buffer.from('{}'),
+    });
+  });
+
+  it('should support `--context` param with `--raw` param', async () => {
+    const lambdaInvokeStub = sinon.stub();
+
+    const result = await runServerless({
+      fixture: 'invocation',
+      cliArgs: ['invoke', '--function', 'callback', '--raw', '--context', '{"ctx": "somecontext"}'],
+      awsRequestStubMap: {
+        Lambda: {
+          invoke: args => {
+            lambdaInvokeStub.returns('payload');
+            return lambdaInvokeStub(args);
+          },
+        },
+      },
+    });
+    expect(lambdaInvokeStub).to.have.been.calledOnce;
+    expect(lambdaInvokeStub.args[0][0]).to.deep.equal({
+      ClientContext: 'IntcImN0eFwiOiBcInNvbWVjb250ZXh0XCJ9Ig==', // "{\"ctx\": \"somecontext\"}"
+      FunctionName: result.serverless.service.getFunction('callback').name,
+      InvocationType: 'RequestResponse',
+      LogType: 'None',
+      Payload: Buffer.from('{}'),
+    });
+  });
+
+  it('should support `--contextPath` param', async () => {
+    const lambdaInvokeStub = sinon.stub();
+
+    const tmpDirPath = getTmpDirPath();
+    fse.mkdirSync(tmpDirPath);
+    const data = {
+      testProp: 'testValue',
+    };
+    const contextDataFile = path.join(tmpDirPath, 'context.json');
+    fse.writeFileSync(contextDataFile, JSON.stringify(data));
```

Comment:
Instead of writing file, let's add `context.json` file to the `invocation` fixture (we already have there `payload.json` etc.)